packages/time-input/src/utils/convert12hTo24h/convert12hTo24h.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Converts a 12 hour format hour to a 24 hour format hour
+ *
+ * @example
+ * ```js
+ * convert12hTo24h('12', 'AM'); // '0'
+ * convert12hTo24h('12', 'PM'); // '12'
+ * convert12hTo24h('1', 'AM'); // '1'
+ * convert12hTo24h('1', 'PM'); // '13'
+ * ```
+ *
+ * @param hour - The hour to convert
+ * @param dayPeriod - The day period to use for the conversion (AM or PM)
+ * @returns The converted hour
+ */
+export const convert12hTo24h = (hour: string, dayPeriod: string) => {
```

Comment:
Why is `hour: string` , and not `number`?

Reply:
Also, should `dayPeriod` be an enum or string literal?

```suggestion
export const convert12hTo24h = (hour: string, dayPeriod: 'AM' | 'PM') => {
```